packages/ag-charts-community/src/chart/series/cartesian/abstractBarSeries.ts

```diff
@@ -88,18 +87,11 @@ export abstract class AbstractBarSeries<
             : xAxis.scale.bandwidth;
     }
 
-    protected getXRangeInVisibleRange(visibleRange: [number, number]): [number, number] | undefined {
-        const { dataModel, processedData } = this;
-        if (!dataModel || !processedData || this.getBarDirection() === ChartAxisDirection.X) return;
-
+    override xCoordinateRange(xValue: any): [number, number] {
         const xScale = this.axes[ChartAxisDirection.X]!.scale;
-        const xValues = dataModel.resolveKeysById(this, `xValue`, processedData);
         const barWidth = xScale.bandwidth ?? 0;
-
-        return visibleRangeIndices(xValues.length, visibleRange, (index) => {
-            const x = xScale.convert(xValues[index]);
-            return [x, x + barWidth];
-        });
+        const x = xScale.convert(xValue);
+        return [x, x + barWidth];
```

Comment:
Q: Shouldn't this be `[x - barWidth/2, x + barWdith/2]`?

Reply:
I believe not - but I do have some off-by-one errors on these that I want to fix in an immediate follow up. Because it also affects the auto-scaling, I'd rather do it in its own piece

Reply:
SGTM